include/boost/typeof/typeof.hpp

```diff
@@ -15,7 +15,7 @@
 
 #include <boost/config.hpp>
 
-#if !defined(BOOST_NO_CXX11_DECLTYPE) && !defined(BOOST_NO_CXX11_TEMPLATE_ALIASES) && !defined(BOOST_TYPEOF_EMULATION)
+#if (!defined(BOOST_MSVC) || (BOOST_MSVC > 1900)) && !defined(BOOST_NO_CXX11_DECLTYPE) && !defined(BOOST_NO_CXX11_TEMPLATE_ALIASES) && !defined(BOOST_TYPEOF_EMULATION)
```

Comment:
Should probably use `&& !BOOST_WORKAROUND(BOOST_MSVC, <= 1900)` here, as that's what it is.